docs/root/configuration/other_features/internal_listener.rst

```diff
@@ -39,85 +55,46 @@ This extension emits the following statistics:
 
    no_metadata, Counter, Metadata key is absent from the import location.
 
-Example config
---------------
-Below is a smallest static config that redirect TCP proxy on port 19000 to the TCP proxy binding to the internal address.
-
-.. code-block:: yaml
-
-  static_resources:
-    listeners:
-    - name: outbound_tcp_svc_19000
-      address:
-        socket_address:
-          address: 0.0.0.0
-          port_value: 19000
-      filter_chains:
-      - filters:
-        - name: tcp_proxy
-          typed_config:
-            "@type": type.googleapis.com/envoy.extensions.filters.network.tcp_proxy.v3.TcpProxy
-            cluster: bridge_internal_listener
-            stat_prefix: svc_tcp_proxy
-    - name: singleton_internal_encap
-      address:
-        envoy_internal_address:
-          server_listener_name: singleton_internal_encap
-      filter_chains:
-      - filters:
-        - name: tcp_proxy
-          typed_config:
-            "@type": type.googleapis.com/envoy.extensions.filters.network.tcp_proxy.v3.TcpProxy
-            cluster: singleton_internal_encap
-            stat_prefix: encap_tcp_proxy
-    clusters:
-    - name: bridge_internal_listener
-      connect_timeout: 3600s
-      type: STATIC
-      load_assignment:
-        cluster_name: "bridge_internal_listener"
-        endpoints:
-        - lb_endpoints:
-          - endpoint:
-              address:
-                envoy_internal_address:
-                  server_listener_name: singleton_internal_encap
-      transport_socket:
-        name: envoy.transport_sockets.raw_buffer
-        typed_config:
-          "@type": type.googleapis.com/envoy.extensions.transport_sockets.raw_buffer.v3.RawBuffer
-    - name: singleton_internal_encap
-      connect_timeout: 3600s
-      type: STATIC
-      load_assignment:
-        cluster_name: "singleton_internal_encap"
-        endpoints:
-        - lb_endpoints:
-          - endpoint:
-              address:
-                socket_address:
-                  address: 0.0.0.0
-                  port_value: 19001
-  bootstrap_extensions:
-  - name: envoy.bootstrap.internal_listener
-    typed_config:
-      "@type": "type.googleapis.com/envoy.extensions.bootstrap.internal_listener.v3.InternalListener"
-
-Real world use cases
---------------------
-
-Encap HTTP GET requests in a HTTP CONNECT request
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Endpoint disambiguation
+-----------------------
+
+In case there are multiple endpoints referencing the same internal listener in a single upstream cluster, use
+:ref:`endpoint ID field <envoy_v3_api_field_config.core.v3.EnvoyInternalAddress.endpoint_id>` to improve change tracking
+in the cluster pool. This field in combination with the internal listener name uniquely identify an endpoint in the
```

Comment:
What happens if you don't use the endpoint id in case? What do you end up losing out on? Or does it just not work?

Reply:
It'll de-duplicate the internal addresses. The same happens for IP:PORTs, but not well-documented. This is due to `HostMap` using strings as a key for host tracking. Should we add this comment here?

Reply:
I think it would be helpful to either make the language prescriptive (you should do this when multiple IPs to ensure X) or help the reader understand what this does if we want to talk about this as an optional thing to do

Reply:
Done.